server/middlewares/chefProfileMiddlewares.js

```diff
@@ -0,0 +1,36 @@
+const ChefProfile = require("../models/chefProfileModel");
+
+const getChefProfile = async (req, res, next) => {
+  try {
+    console.log("get chef profile middleware reached");
+
+    //get userId from middleware
+    const userId = req.user._id;
+
+    // find profile by user._id
+    const chefProfile = await ChefProfile.findOne()
+      .populate({
+        path: "user",
+        match: {
+          _id: userId,
+        },
+        select: "-password", // DO NOT INCLUDE PASSWORD
+      })
+      .exec();
+
+    if (chefProfile.length < 1) throw "Profile not found";
+
+    // save the chefProfile inside the req.chefProfile
+    req.chefProfile = chefProfile;
+
+    next();
+  } catch (error) {
+    console.log(error);
+
+    return res.status(404).json({ error });
```

Comment:
For a 404 error it would be better to handle that specifically rather than catching a general error. Maybe instead of throwing "Profile not found" you can send the res then and change this line to a more general 503 error

Reply:
I'm not sure if I fix this correctly.

`const exists = await Model.query`
`if (!exists) throw new Error(exists)` or
`if (!exists) throw new Error("Not found")`

Do you want me to pass the query as params then catch return res.status(503).json({error:error.message})?

Reply:
If the document is not found, you can just put
```javascript
return res.status(404).json({message: "_ not found"})
```

Reply:
I fixed it. Thank you.